### PR TITLE
[WIP] Resolution of default city

### PR DESCRIPTION
**Pull request by mukarramali**
Issue to be resolved: #6

### CODE REVIEW

**Comment by mukarramali** [comment]
Now we are not taking latlng as the basis. Instead using, current political city name.
Run rake task to update previous default cities.
`$ rake location_fixes:update_default_cities`

**Comment by shireeshj** [comment]
What is the status of this PR?

File: app/models/autocomplete_locations/autocomplete_location.rb [hunk 1]
```diff
@@ -12,5 +12,23 @@ def self.city_name_ids(id)
...
+    end
+
+    def self.default_city(id)
+      default_city = AutocompleteLocation.find_by(city_id: AutocompleteLocation.find(id).city_id, default_city: true)
+      default_city || AutocompleteLocation.find(id)
+    end
+
+    def self.current_city_name?(query)
+      # Try query with city name and state both. Ex: query = 'Aurangabad, Uttar Pradesh'
+      searched_city = Geocoder.search query
```
Comment (bitsapien) [inline app/models/autocomplete_locations/autocomplete_location.rb]: Might need to consider hard dependency on `Geocoder` here for this task specifically.

Comment (bitsapien) [reply app/models/autocomplete_locations/autocomplete_location.rb]: Also, this change was specifically required for our DB, might not be relevant generally. I would suggest looking at it and closing this PR is irrelevant.

Comment (shireeshj) [reply app/models/autocomplete_locations/autocomplete_location.rb]: Does it not also handle cases where a city was renamed?

Comment (mukarramali) [reply app/models/autocomplete_locations/autocomplete_location.rb]: yes, this has hard dependency on `geocoder`, and I found out they keep changing their API.
I'll see what else can be the solution for finding out which should be the current default city.
Like how to decide if `Bangalore` or `Bengaluru` should be used.

**Pull request closed without merging**